In [21]:
import pandas as pd
import numpy as np
import os
import seaborn as sns
import matplotlib.pyplot as plt

# Data Understanding

In [13]:
links = pd.read_csv('../ml-32m/links.csv')
movies = pd.read_csv('../ml-32m/movies.csv')
ratings = pd.read_csv('../ml-32m/ratings.csv')
tags = pd.read_csv('../ml-32m/tags.csv')

#### ../ml-32/something the dots mean that you have to go one directory upwards

## links
- 127 Nullwerte bei tmdbId

In [14]:
links.head(5)

movieId  imdbId   tmdbId
0        1  114709    862.0
1        2  113497   8844.0
2        3  113228  15602.0
3        4  114885  31357.0
4        5  113041  11862.0

In [15]:
links.count()

movieId    87585
imdbId     87585
tmdbId     87461
dtype: int64

In [24]:
nulls = pd.DataFrame(links.isna().sum())

In [26]:
nulls

0
movieId    0
imdbId     0
tmdbId   124

## movies
- viele genres in einer Spalte
- Jahr in einer Spalte zusammen mit Titel
- keine NUllwerte
- 87585 Zeilen

In [31]:
movies.head(5)

movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [32]:
movies.count()

movieId    87585
title      87585
genres     87585
dtype: int64

## ratings
- keine Nullwerte
- avg Rating: 3.54
- max Rating: 5.0
- min Rating: 0.5
- timestamp?

In [33]:
ratings.head(5)

userId  movieId  rating  timestamp
0       1       17     4.0  944249077
1       1       25     1.0  944250228
2       1       29     2.0  943230976
3       1       30     5.0  944249077
4       1       32     5.0  943228858

In [34]:
ratings.count()

userId       32000204
movieId      32000204
rating       32000204
timestamp    32000204
dtype: int64

In [35]:
ratings.isna().sum()

userId       0
movieId      0
rating       0
timestamp    0
dtype: int64

In [36]:
ratings['rating'].describe()

count    3.200020e+07
mean     3.540396e+00
std      1.058986e+00
min      5.000000e-01
25%      3.000000e+00
50%      3.500000e+00
75%      4.000000e+00
max      5.000000e+00
Name: rating, dtype: float64

## tags
- 140980 verschiedene Tags
- 17 Nullwerte
- es gibt 2 Milionen Zeilen, nicht jedes Film hat aber ein Tag sondern nur 51323 was ca. 5/8 ist

In [37]:
tags.head(5)

userId  movieId          tag   timestamp
0      22    26479  Kevin Kline  1583038886
1      22    79592     misogyny  1581476297
2      22   247150   acrophobia  1622483469
3      34     2174        music  1249808064
4      34     2174        weird  1249808102

In [40]:
len(tags['tag'].unique())

140980

In [41]:
tags.count()

userId       2000072
movieId      2000072
tag          2000055
timestamp    2000072
dtype: int64

In [42]:
tags.isna().sum()

userId        0
movieId       0
tag          17
timestamp     0
dtype: int64

In [44]:
len(tags['movieId'].unique())

51323